src/index.js

```diff
@@ -0,0 +1,284 @@
+import StatusApi from './statusApi.js';
+import {buildUrl} from './utils';
+const Ip = require('ip');
+const fs = require('fs');
+const path = require('path');
+
+// Duration (in ms) to wait before attempting to reconnect.
+const CONNECT_INTERVAL = 4000;
+const CHAIN_NAME = 'embark';
+const NETWORK_NAME = 'Embark';
+
+// private method symbols
+const _connectStatus = Symbol('connectStatus');
+const _connectToStatusNetwork = Symbol('connectToStatusNetwork');
+const _addAndConnectStatusNetwork = Symbol('addAndConnectStatusNetwork');
+const _getStatusNetworkId = Symbol('getStatusNetworkId');
+const _setStatusNetworkId = Symbol('setStatusNetworkId');
+const _registerEvents = Symbol('registerEvents');
+
+/**
+ * Plugin that connects an Embark dApp to the Status app, and allows the dApp
+ * to be run in the Status browser.
+ */
+class EmbarkStatusPlugin {
+  constructor(embark) {
+    this.networkId = 1337;
+    this.embark = embark;
+    this.events = this.embark.events;
+    this.pluginConfig = this.embark.pluginConfig;
+    this.deviceIp = this.pluginConfig.deviceIp;
+    this.logger = this.embark.logger;
+    // we want a timeout to occur before issuing the next request, so reduce the timeout specified
+    this.statusApi = new StatusApi({ip: this.deviceIp, logger: this.logger, timeout: CONNECT_INTERVAL - 1000});
+    this.webServerConfig = {};
+    this.blockchainConfig = {};
+    this.networkIdDirectory = path.join(process.env.DAPP_PATH, '.embark', 'embark-status');
+    this.networkIdPath = path.join(this.networkIdDirectory, 'networkId');
+
+    // gets hydrated webserver config from embark
+    this.events.on('config:load:webserver', webServerConfig => {
```

Comment:
Can't you just use `embark.config.webserverConfig`?

Reply:
Because plugins are instantiated first in the lifecycle of Embark, these configs are not hydrated properly yet inside of the plugin constructor. Originally, to workaround this, I had them being read from `embark.config` every time the library needed them, but I figured that if we run in to race conditions with the configs not being set yet when they're needed, at least this pattern with events allows us a way to mitigate the race if needed.

Reply:
Understood